### PR TITLE
Change GridSpace -> Grid and SubSet -> Domain

### PR DESCRIPTION
**Pull request by guberger**
This PR changes the names of the structure `GridSpace` and `SubSet` in `Grid` and `Domain`, respectively. Also `Grid` now contains only the information to make the mapping between `pos` and `coord` (i.e., only `orig` and `h`). That is, a set of cells now must be defined via a `Domain` (thus the functions `add_pos!`, `add_set!`, etc. apply exclusively on `Domain`), and `SymbolicModel` needs `Domain`'s (and not `GridSpace`'s) as fields.

### CODE REVIEW

File: src/automaton.jl [hunk 1]
```diff
@@ -24,13 +24,20 @@ function get_ntrans(autom::AutomatonList)
     return length(autom.transitions)
 end
 
+# In add_trans and add_translist:
 # Do not check that source, symbol, target are "inbounds"
 # Assumes not add twice same transition...
-function add_transition!(autom::AutomatonList, source, symbol, target)
+function add_trans!(autom::AutomatonList, source, symbol, target)
```
Comment (blegat) [inline src/automaton.jl:30]: Add some point, we might want to merge the interface with the one in HybridSystems so we'd rather stay close:
https://blegat.github.io/HybridSystems.jl/latest/lib/methods/#HybridSystems.add_transition!

File: src/domain.jl [hunk 2]
```diff
@@ -0,0 +1,104 @@
+abstract type Domain{N,T} end
+
+# Without S, add_set! and remove_set! where not type-stable...
+struct DomainList{N,T,S<:Grid{N,T}} <: Domain{N,T}
+    grid::S
+    elems::Set{NTuple{N,Int}}
+end
+
+function NewDomainList(grid::S) where {N,T,S<:Grid{N,T}}
```
Comment (blegat) [inline src/domain.jl:9]: Why not create a classical constructor `DomainList(grid::S)` ?

File: src/grid.jl [hunk 3]
```diff
@@ -0,0 +1,43 @@
+abstract type Grid{N,T} end
+
+# Free because later: maybe put bounds lb and ub (e.g., for BDDs)
+struct GridFree{N,T} <: Grid{N,T}
+    orig::SVector{N,T}
+    h::SVector{N,T}
+end
+
+function NewGrid(orig::SVector{N,T}, h::SVector{N,T}) where {N,T}
```
Comment (blegat) [inline src/grid.jl:9]: Why not call directly `GridFree` ? What is the use of `NewGrid` ?

**Comment by blegat** [comment]
It looks much cleaner indeed

File: src/grid.jl (hunk 3 above)
Comment (guberger) [reply src/grid.jl]: Changed

File: src/domain.jl (hunk 2 above)
Comment (guberger) [reply src/domain.jl]: changed

File: src/automaton.jl (hunk 1 above)
Comment (guberger) [reply src/automaton.jl]: changed

**Comment by guberger** [comment]
Results are similar: with `example_pathplanninggrowth-hard`:
* Before:
```
compute_symmodel_from_controlsystem! started
compute_symmodel_from_controlsystem! terminated with success: 23784452 transitions created
  7.586279 seconds (26 allocations: 2.141 KiB)
compute_controller_reach! started
compute_controller_reach! terminated with success
  0.697363 seconds (65 allocations: 25.924 MiB)
```
* After:
```
compute_symmodel_from_controlsystem! started
compute_symmodel_from_controlsystem! terminated with success: 23784452 transitions created
  7.568163 seconds (23 allocations: 2.078 KiB)
compute_controller_reach! started
compute_controller_reach! terminated with success
  0.688603 seconds (62 allocations: 25.924 MiB)
```

**Pull request merged**